CONTRIBUTING.md

```diff
@@ -1,5 +1,132 @@
+
+As you might imagine web code is very easy to test and contribute to, so that's all on the table! You may find that some of the web features you're looking for are actually in the [expo-cli repo](https://github.com/expo/expo-cli).
+
+## 📦 Download and Setup
+
+1. [Fork](https://help.github.com/articles/fork-a-repo/) this repository to your own GitHub account and then [clone](https://help.github.com/articles/cloning-a-repository/) it to your local device. (`git remote add upstream git@github.com:expo/expo.git` 😉)
+2. Run the setup script with: `npm run setup`
+
+   <!-- TODO(Bacon): Split this into 2 scripts so people can contribute to docs without installing React Native -->
+
+   - Downloads submodules (like `react-native`) with `git submodule update --init`
+   - Fetches files with [_git lfs_](https://git-lfs.github.com), which we use for big native libraries like Google Mobile Vision. Note: you must have `git lfs` already installed.
+   - Ensures Yarn is installed
+   - Ensures your computer is set up for React Native (will install the Android NDK if it's not present)
+   - Downloads the Node packages (`yarn install`)
+
+3. Navigate to the bare sandbox project `cd apps/bare-expo`
+4. Run the project on any platform (maybe start with web; it's the fastest! 😁)
+
+   - Web: `yarn web`
+   - iOS: `yarn ios`
+   - Android: `yarn android`
+
+> If this didn't work for you as described, please [open an issue.](https://github.com/expo/expo/issues/new/choose)
+
+## ✏️ Editing Packages
+
+All of our packages (including Foundation Unimodules) can be found in the `packages/` directory. These packages are universally linked to the projects in the `apps/` directory (meaning any iOS, Android, web, or API changes can be tested from `apps/bare-expo/`). `bare-expo` is a bare-workflow Expo app that acts as a runner for the other projects in the `apps/` directory.
+
+- `native-component-list`: This is where you can write demos or tests that require physical interaction (a good playground for testing).
+- `test-suite`: You can write your E2E tests in here. When pushed to the remote, CI will run this project with Device Farm for Android, Detox for iOS, and Puppeteer for web!
+
+1. Navigate to a package you want to edit. Ex: `cd packages/expo-constants`
+2. Start the TypeScript build in watch mode: `yarn build`
+3. Edit code in that package's `src/` directory
+4. Play with your changes on a simulator or device through `bare-expo`:
+   - Add or modify a file named after the API you're working on. Ex: `apps/test-suite/tests/Constants.js`
+   - Run the code from the `bare-expo` project with `yarn <android | ios | web>` or test the code with `yarn test:<ios | web>`
+5. You can edit the native code by opening the projects in their respective editors:
```

Comment:
might be worth clarifying here that you can edit the ios, android, and js files directly in the packages directory to test the changes